### PR TITLE
In Overlay do not call API directly since it does not have access to the token auth.

### LINKED ISSUE #13406: Error: You can't access this resource as it requires 'view' access for the website id = 60.
I'm using Page Overlay for one of our sites and have super user permissions for my user. Therefore I don't understand, why I can't see these bubbles on the page overlay. Everthing else (like Open Row Evolution, Open Transitions, Open segmented Visitor Log) works fine.

Both sites (Matomo + Website) use TLS. But I only get the page, this error and after clicking ok in the right corner: "Loading following pages ..."

I enabled debug mode and got this output:
<pre>
DEBUG Piwik\Plugin\Manager[2018-09-10 15:33:36 UTC] [43c7b] Loaded plugins: CorePluginsAdmin, CoreAdminHome, CoreHome, WebsiteMeasurable, Diagnostics, CoreVisualizations, Proxy, API, Widgetize, Transitions, LanguagesManager, Actions, Dashboard, MultiSites, Referrers, UserLa
nguage, DevicesDetection, Goals, Ecommerce, SEO, Events, UserCountry, GeoIp2, VisitsSummary, VisitFrequency, VisitTime, VisitorInterest, RssWidget, Monolog, Login, UsersManager, SitesManager, Installation, CoreUpdater, CoreConsole, ScheduledReports, UserCountryMap, Live, C
ustomVariables, PrivacyManager, ImageGraph, Overlay, SegmentEditor, Insights, Morpheus, Contents, Resolution, DevicePlugins, Heartbeat, Intl, Marketplace, ProfessionalServices, UserId, CustomPiwikJs, DBStats, Provider, Bandwidth, CustomOptOut, LogViewer, SecurityInfo, Site
Migration, TasksTimetable                                                                                                                                                                                                                                                        
DEBUG API[2018-09-10 15:33:36 UTC] [43c7b] /var/www/matomo/core/Access.php(505): You can't access this resource as it requires 'view' access for the website id = 60.                                                                                                            
DEBUG API[2018-09-10 15:33:36 UTC] [43c7b] #0 /var/www/matomo/core/Piwik.php(511): Piwik\Access->checkUserHasViewAccess('60')                                                                                                                                                    
DEBUG API[2018-09-10 15:33:36 UTC] [43c7b] #1 /var/www/matomo/plugins/SitesManager/API.php(223): Piwik\Piwik::checkUserHasViewAccess('60')                                                                                                                                       
DEBUG API[2018-09-10 15:33:36 UTC] [43c7b] #2 /var/www/matomo/plugins/Overlay/API.php(47): Piwik\Plugins\SitesManager\API->getSiteFromId('60')                                                                                                                                   
DEBUG API[2018-09-10 15:33:36 UTC] [43c7b] #3 [internal function]: Piwik\Plugins\Overlay\API->getExcludedQueryParameters('60')                                                                                                                                                   
DEBUG API[2018-09-10 15:33:36 UTC] [43c7b] #4 /var/www/matomo/core/API/Proxy.php(236): call_user_func_array(Array, Array)                                                                                                                                                        
DEBUG API[2018-09-10 15:33:36 UTC] [43c7b] #5 /var/www/matomo/core/API/Request.php(262): Piwik\API\Proxy->call('\\Piwik\\Plugins\\...', 'getExcludedQuer...', Array)                                                                                                             
DEBUG API[2018-09-10 15:33:36 UTC] [43c7b] #6 /var/www/matomo/plugins/API/Controller.php(41): Piwik\API\Request->process()                                                                                                                                                       
DEBUG API[2018-09-10 15:33:36 UTC] [43c7b] #7 [internal function]: Piwik\Plugins\API\Controller->index()                                                                                                                                                                         
DEBUG API[2018-09-10 15:33:36 UTC] [43c7b] #8 /var/www/matomo/core/FrontController.php(560): call_user_func_array(Array, Array)                                                                                                                                                  
DEBUG API[2018-09-10 15:33:36 UTC] [43c7b] #9 /var/www/matomo/core/FrontController.php(146): Piwik\FrontController->doDispatch(NULL, NULL, NULL)                                                                                                                                 
DEBUG API[2018-09-10 15:33:36 UTC] [43c7b] #10 /var/www/matomo/core/dispatch.php(34): Piwik\FrontController->dispatch()                                                                                                                                                          
DEBUG API[2018-09-10 15:33:36 UTC] [43c7b] #11 /var/www/matomo/index.php(27):require_once('/var/www/matomo...')                                                                                                                                                                  
DEBUG API[2018-09-10 15:33:36 UTC] [43c7b] #12 {main}  
</pre>

I'm quite lost right now. Can somebody help me? :)

### PR DESCRIPTION
**Pull request by diosmosis**
Fixes #13406 

Authenticating via UI session in API requests is no longer supported.

### CODE REVIEW

File: plugins/Overlay/Controller.php [hunk 1]
```diff
@@ -235,4 +251,10 @@ protected function outputCORSHeaders()
         $corsHandler = new CORSHandler();
         $corsHandler->handle();
     }
+
+    private function apiRequest($methodName)
+    {
+        $_GET['method'] = $_POST['method'] = 'Overlay.' . $methodName;
```
Comment (tsteur) [inline plugins/Overlay/Controller.php]: is it possible to use `Request::processRequest` instead of dispatching new request fully?

Comment (diosmosis) [reply plugins/Overlay/Controller.php]: I could use `new Request(...)` (processRequest hard codes some params)..

**Comment by diosmosis** [comment]
Updated.

**Comment by diosmosis** [comment]
Closing in favor of https://github.com/matomo-org/matomo/pull/13446

**Pull request closed without merging**